Update readme to include conda env config vars option

AndrewAnnex opened the pull request:
Updates readme step 7 of install to include steps for using `conda env
config vars` to set environment variables as option 2.

## Description
This is the first PR to address the feature discussed in issue #4499, currently this is a work in progress as there are possibly other documentation areas to update.
 
There are also a few questions I have that need answered and may require more changes to this pr:

1. ~~Is ISISTESTDATA actually required?~~
2. ~~What about ALEDATA?~~
3. Update miniconda/anaconda version mentioned to a more recent release
4. ~~Can I remove mentions of isis version < 4.2.0, as older readmes contain
   install directions.~~
5. Verify I got the update command correct.
6. See why the markdown doesn't indent for options 1 and 2.

## Related Issue
Issue #4499 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
method tested locally by myself, and in CI jobs for asap-stereo.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- - [ ] My code follows the code style of this project. -->
- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

AndrewAnnex committed:
Update readme to include conda config var option

First step for issue #4499

Updates readme step 7 of install to include steps for using `conda env
config vars` to set environment variables as option 2.

there are a few things to work out:

1. Is ISISTESTDATA actually required?
2. What about ALEDATA?
3. Update miniconda/anaconda version mentioned to a more recent release
4. Remove mentions of isis version < 4.2.0, as older readmes contain
   install directions
5. Verify I got the update command correct.
6. See why the markdown doesn't indent for options 1 and 2.

jessemapel commented:
>     1. Is ISISTESTDATA actually required?

This is only needed to run the Makefile based app and integration tests. So, it's really not needed but we keep it in there because the old setup script would set it up.

>     2. What about ALEDATA?

This is definitely not needed. This is used by ALE to generate ISDs outside of ISIS and was previously required to supress a warning but is no longer.

>     4. Can I remove mentions of isis version < 4.2.0, as older readmes contain
>        install directions.

I would add a note at the top of the section that says these are only valid for 4.2+ and a link to the older README. I can find the right README to link to.

AndrewAnnex committed:
Update README.md

removing direct mentions of ISIS 4.2.0 from env steps, added isis 5.0.0 readme link.

AndrewAnnex commented:
@jessemapel okay I think I accidentally "dismissed" the review by making the edits directly in the pr. I've added a link to the ISIS 5.0.0 Readme file for versions of isis older than 4.2.0 which cleaned up a few things.

I actually think I should remove the mentions of ISISTESTDATA, it is pretty confusing to me and to newcomers (I think) that you don't actually need the testdata or the testdata environment variable to use isis. The testdata should only really be relevant to devs/ people interested in compiling ISIS so I think any mention of it should be moved to those wiki pages/etc. 

As long as ISIS doesn't have a runtime dependency on the environment variable it shouldn't be set to a dummy path. Of course this also simplifies the readme further

rbeyer commented on README.md:
```diff
@@ -91,30 +91,63 @@ This installation guide is for ISIS users interested in installing ISIS (3.6.0)+
+      This version number applies only to the conda package, not to the version of miniconda or anaconda that was installed. 
 
-        python $CONDA_PREFIX/scripts/isis3VarInit.py --data-dir=[path to data directory]  --test-dir=[path to test data directory]
+      To determine if your version of conda is recent enough run:
```
I appreciate the verbosity, but do we really need to lay out how to check your conda version and how to update it?  Seems like that reproduces things that the conda docs provide.

jessemapel replied:
We've provided some of this to help users get used to using conda to install instead of just rsync. I would be fine with just a link to the conda docs. For this PR, I think everything is totally fine.

AndrewAnnex replied:
I agree with leaving it, I suspect there is a non-zero population of isis users who only use conda for ISIS. I could add a link in the docs as well but I also think this is sufficient for now.

jessemapel reviewed (dismissed):
There is no runtime dependency on ISISTESTDATA so we can remove it. I'll put in an issue to remove it from the user docs and the environment setup stuff.

Also, pushing more commits to a PR that has been reviewed will auto-dismiss the reviews so that it has to be re-reviewed.

AndrewAnnex committed:
remove direct mention of ISISTESTDATA from readme

AndrewAnnex commented:
@jessemapel PR has been updated to remove direct mentions of ISISTESTDATA.

jessemapel reviewed (approved):
I think this is ready to go. I'll get a PR in today to remove ISISTESTDATA from the activate scripts too.
